### PR TITLE
xla_lhlo.neg and xla_lhlo.abs support integer arguments

### PR DESCRIPTION
**Pull request by qqsun8819**
Do self transformation based on current mlir standard dialect

### CODE REVIEW

**Comment by qqsun8819** [comment]
@joker-eph  plz take a look ^_^

File: tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h [hunk 1]
```diff
@@ -119,8 +124,24 @@ template <>
...
+  if (element_type.isa<IntegerType>()) {
+    const auto& lhs = args[0];
+    auto integer_type = element_type.dyn_cast<IntegerType>();
+
+    auto zero_intval = b->create<::mlir::ConstantIntOp>(loc, 0, integer_type.getWidth());
+    auto lhs_gt_zero = b->create<ScalarIOp<CompareOp>>(
+        loc, CmpIPredicate::sge, lhs, zero_intval);
+    auto neg_val =  b->create<ScalarIOp<xla_lhlo::NegOp>>(
+        loc, result_types,
+        args, mlir::None);
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h]: Seems like we can just use `sub(zero_intval, lhs)` instead of the NegOp?

**Comment by gbaned** [comment]
@qqsun8819  Can you please check joker-eph's comments and resolve conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

File: tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h (hunk 1 above)
Comment (qqsun8819) [reply tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h:141]: Ok mehdi ,I'll update it

**Comment by qqsun8819** [comment]
I'll update this PR according to what @joker-eph comment

**Comment by qqsun8819** [comment]
@joker-eph please take a look again ^_^

File: tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h [hunk 2]
```diff
@@ -119,8 +119,23 @@ template <>
...
     OpBuilder* b) {
-  return MapLhloOpToStdScalarOpImpl<FloatType, ::mlir::AbsFOp>{}(
-      loc, result_types, args, b);
+  Type element_type = args.front().getType();
+  if (element_type.isa<FloatType>()) {
+    return MapLhloOpToStdScalarOpImpl<FloatType, ::mlir::AbsFOp>{}(
+        loc, result_types, args, b);
+  }
+  if (element_type.isa<IntegerType>()) {
+    const auto& lhs = args[0];
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h:128]: Nit: `Value lhs = args[0]`

Also can you add a comment on the expansion at the beginning of the block? Something like:
```
// lhlo.abs(x, y) ->  select((x>0), x, (0-y))
```

File: tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h [hunk 3]
```diff
@@ -326,8 +341,22 @@ template <>
...
-  return MapLhloOpToStdScalarOpImpl<FloatType, ::mlir::NegFOp>{}(
-      loc, result_types, args, b);
+  Type element_type = args.front().getType();
+  if (element_type.isa<FloatType>()) {
+    return MapLhloOpToStdScalarOpImpl<FloatType, ::mlir::NegFOp>{}(
+        loc, result_types, args, b);
+
+  }
+  if (element_type.isa<IntegerType>()) {
+    const auto& lhs = args[0];
```
Comment (joker-eph) [inline tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h:351]: (same)

File: tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h (hunk 2 above)
Comment (qqsun8819) [reply tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h:128]: fixed

File: tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h (hunk 3 above)
Comment (qqsun8819) [reply tensorflow/compiler/mlir/xla/transforms/map_xla_to_scalar_op.h]: fixed

**Comment by qqsun8819** [comment]
@joker-eph pr updated according to you comments, take a look again ^_^

**Review by joker-eph — APPROVED** [review]

**Pull request merged**